src/control/server/init/setup_spdk.sh

```diff
@@ -17,12 +17,9 @@ else
 	PCI_WHITELIST="$_PCI_WHITELIST" NRHUGE="$_NRHUGE" \
 	TARGET_USER="$_TARGET_USER" "$scriptpath"
 
-	eet -e
-
+	# ignore and suppress errors from missing UIO related files
 	chown -R "$_TARGET_USER" /dev/hugepages /dev/uio* \
 		/sys/class/uio/uio*/device/config \
-		/sys/class/uio/uio*/device/resource*
-
-	set e
+		/sys/class/uio/uio*/device/resource* 2>/dev/null || true
```

Comment:
Do you really want to ignore an error status returned by `chown` and continue on?  Isn't it likely that the failure to change ownership is going to result in a failure at some future step when it will be more difficult to determine that it was the `chown` failure causing the problem?

If the issue is that some of the items you are `chown`ing might not exist, test for their existence and only include them in `chown` argument list if they exist or expand the glob before giving it to `chown` with something like:

```bash
chown  -R "$_TARGET_USER" /dev/hugepages $(ls /dev/uio*) \
	$(ls /sys/class/uio/uio*/device/config) \
	$(ls /sys/class/uio/uio*/device/resource*) 2>/dev/null || true
```

The caveat with the above is that you might blow the arglist length limit.  In that case you will need to test each glob and build your arg list accordingly.  i.e.:

```bash
arglist="/dev/hugepages"
for glob in '/dev/uio*' '/sys/class/uio/uio*/device/config' '/sys/class/uio/uio*/device/resource*'; do
    if ls "$glob"; then
        arglist="$arglist $glob"
    fi
done
chown -R "$_TARGET_USER" "$arglist"
```

Without trying the above, I cannot be sure I've got the quoting correct, but it demonstrates the general idea.

Reply:
I'm afraid I wasn't thinking the above suggestion through entirely.  I was forgetting that tools like `chown` and `ls` don't expand globs themselves but rely on the shell to do it.  So your final `$arglist` ends up being a list of globs that have to be expanded by the shell before giving them to `chown`, which is basically the same problem as the first option above.

I really have no idea how many items those globs could expand to and whether _arglist too long_ is a potential problem or not.  I feel like you might think it could be which is why you opted to implement this second option.  If you really don't think it could ever be a problem, the first option with the `$(ls ...)` commands after the `chown` is probably the most straightforward.

If you do think all of those globs could expand to blow the argument list length (2097152 bytes on EL7 currently -- can be obtained with `getconf ARG_MAX`) then you could do a combination of the two approaches above such as:

```bash
for glob in /dev/hugepages '/dev/uio*' '/sys/class/uio/uio*/device/config' '/sys/class/uio/uio*/device/resource*'; do
    if list=$(ls $glob); then
        echo "$list" | xargs -r chown -R "$_TARGET_USER"
    fi
done
```

That then breaks down your potentially large single expanded arglist and `chown` command down into smaller ones.

My sincerest apologies for misleading you above.